### ISSUE TITLE
Compatibility with Patchbox OS

### ISSUE THREAD

**Issue by vaccinemedia** [vicwomg/pikaraoke#236]
I have a pisound which is a zero latency audio hat for the pi and one of the best distros for it is Patchbox OS which is a dedicated audio distro for the pi which comes with a lot of cool features / modules. The pisound itself has stereo mic inputs and outputs and using patchbox os I am able to plug in a mic and use one of the modules called "modep" to have effects applied to the mic input in realtime. It sounds awesome :)

The issue I have is that after installing piaraoke the video plays but the audio only comes from the built in headphone jack of the pi. I'm attaching a photo of the output after pressing the escape key.

![IMG_0818](https://user-images.githubusercontent.com/10465885/220215981-3f4b9266-1adf-408f-951c-04b69c865b5f.jpeg)

It appears as though pulseaudio is already being used by the system and that running pikaraoke as sudo is clashing.

Maybe the solution is for the python app to use the same user which has created the pulseaudio instance? I have tried running without sudo both in the command line and also after booting to desktop and get the same error which is that the backdrop is not a bmp file. So it cannot work at the moment using Patchbox OS as it boots with "modep" using pulse already using the user "patch".

Is there a way in order to make this setup work? You can install Patchbox OS to work with other USB sound cards for testing. From what I can tell, the bug is not being able to run pikaraoke without sudo. As long as pikaraoke can run without sudo it may work but pygame needs sudo to create the qrcode maybe?

Any help in getting pykaraoke working from the command line on the pi without using sudo would be awesome as I think this is where the issue lies :)

**Comment by vaccinemedia** [comment]
Running without sudo gives this:
![IMG_0819](https://user-images.githubusercontent.com/10465885/220222130-31efaa4c-3b35-4284-b440-46c37e020bb6.jpeg)

**Comment by vicwomg** [comment]
I can't really speak to any specifics about Patchbox, but regarding the sudo stuff:

Delete the qr code png file in the root directory of pikaraoke and you should be able to get past that error. The issue is that you already created the file as sudo, so that file cannot be overwritten. 

However I don't think you'll get much further than that because pygame requires sudo to run from console and render graphics, if I recall correctly..

If installing an X windows desktop environment is a possibility, you can run pikaraoke from within X without sudo because it already has graphics environment.

**Comment by vaccinemedia** [comment]
What you mentioned regarding pygame and needing to be run as sudo did the trick.

What I did was:

First thing was to re-flash Patchbox OS and run the wizard and select the desktop auto login environment. Then I ran apt update and inatalled pikaraoke as per the instructions. I then ran it NOT using sudo and made sure to set the download folder to ~/pikaraoke-songs

This got it to run so I connected to it using my phone and started a song. No sound out of pisound but out of the headphone jack. So I shut the pi down and rebooted and on the second try the audio came out of the pisound. Winner :)

So next I installed modep, then installed the lv2 modules as per instructions here [[Plugins] Only 73 plugins available in modep - #6 by Pranciskus](https://community.blokas.io/t/plugins-only-73-plugins-available-in-modep/4274/6) which removed python3-pygame so I uninstalled the plugin which removed it and re-installed pygame.

Then after another reboot I plugged in the mic and connected to modep, added a reverb and that worked on the mic. Running pikaraoke stlll worked so I was able to sing on top of the backing with effects applied. Awesome sauce :)

The audio from pikaraoke / vlc is a little overpowering though and fights with the mic input. I added a gain effect after the reverb one and it worked sort of but I then had peaking on the vocals. So if it's possible to lower the volume on the backing slightly that would make the vocals stand out more but yeah it works!

The reason I wanted to get it working on commandline was that it can auto run on boot and also I thought the desktop OS would be an added CPU usage issue but it's running on the pi 4 with around 25% usage which is great :)
